### PR TITLE
Add flow compatibility

### PR DESCRIPTION
**Pull request by ghmeier**
I updated the `eslint`  dependency to support correctly parsing flow annotated files, and also had to upgrade the minimum supported `babel-eslint`version due to an issue when parsing flow type definitions ([issue here](https://github.com/eslint/eslint/issues/9767#issuecomment-353840674)).

This will be a major version change since it requires 4.X, but shouldn't introduce problems based on [the migration guide](https://github.com/eslint/eslint/blob/master/docs/user-guide/migrating-to-4.0.0.md).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,14 +30,15 @@
...
   "homepage": "https://github.com/mixmaxhq/eslint-config-mixmax#readme",
   "optionalDependencies": {
-    "babel-eslint": "^7.2.3",
+    "babel-eslint": "^8.2.1",
+    "eslint-plugin-flowtype": "^2.41.0",
     "eslint-plugin-react": "^7.1.0"
   },
   "peerDependencies": {
-    "eslint": ">= 3"
+    "eslint": ">=4.14.0"
```
Comment (spencer-brown) [inline package.json]: can you please update the README with updated dependency installation instructions? (describing the optional `flowtype` dependency and updating the version of eslint that needs to be installed)

separately, it looks like the README is outdated in other ways - mentioning forthcoming configs that have already been added, not including the `jest` config, requiring node >=7.6.0, and describing the project as being at `v0.1` - i'm happy to clean these up after you merge this or feel free if you want to knock them out while you're in here.

File: node/flow.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  parser: 'babel-eslint',
+  extends: 'mixmax/node',
+  plugins: [
+    // Allows correct parsing of flow annotated files.
+    'flowtype'
+  ],
+  rules: {
+    'space-infix-ops': 0,
```
Comment (spencer-brown) [inline node/flow.js]: should this go in our base config? same Q on `no-duplicate-imports`

File: node/flow.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  settings: {
+    // Only use the flow linter for files with `@flow`
+    flowtype: {
+      onlyFilesWithFlowAnnotation: true
+    }
+  },
+  parserOptions: {
+    // Required to support import/export syntax when using types
+    sourceType: 'module',
```
Comment (spencer-brown) [inline node/flow.js]: what if we want to use this server-side with `script`-type `sourceType`?

i'm wondering if we want a base `flow` config that could be mixed with other configs to make eg `flow-node` and `flow-browser`.

File: node/flow.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ecmaFeatures: {
+      experimentalObjectRestSpread: true,
```
Comment (spencer-brown) [inline node/flow.js]: what do you think about breaking out a new config to be used by transpiled, node-run code and `extend`ing that config here? here and in my previous comment, just trying to think about how we can best keep these configs DRY - we might want several experimental features in multiple locations, but won't want to copy/paste those everywhere.

here, maybe we want to combine two different configs like `flow` and `node-es6` to create the final product you'd use in a flow-using, transpiled, server-side shared module.

File: node/flow.js (hunk 3 above)
Comment (ghmeier) [reply node/flow.js]: To the first point, this works with server-side code just fine as far as I can tell (https://github.com/mixmaxhq/event-logging/pull/1 this correctly enforces rules using this config). On the other point, we could not extend `mixmax/node` here and instead rely on the client to extend it like `"extends": [ "mixmax/node", "mixmax/flow" ]`

File: node/next.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (spencer-brown) [inline node/next.js]: sweet

**Comment by spencer-brown** [comment]
thanks! excited about this new pattern :).

**Pull request merged**